adr/001-component-library.md

```diff
@@ -0,0 +1,60 @@
+# Introduce component library
+
+- Status: Proposed
+- Deciders: Team
+- Date: 17.10.2024
+
+## Result
+
+A1: Component library is introduced, firstly as just a folder in our current setup, adding yarn workspaces or similar requires more research and testing
+
+## Problem context
+
+Today our UI components are tightly coupled to the app in which they are rendered.
+
+This leads to several issues:
+
+- Makes it hard to do component testing outside a fully rendered app.
+- Makes refactoring the app framework a lot harder.
+- Leads to unclear interfaces between UI components and the app framework.
+- Makes developing UI components complex without deep understanding of the application.
+
+## Decision drivers
+
+A list of decision drivers. These are points which can differ in importance. If a point is "nice to have" rather than
+"need to have", then prefix the description.
+
+- B1: UI components should only receive data to display, notify the app when data is changed, and notify validity of user input.
```

Comment:
```suggestion
- B1: UI components should only receive data to display, and notify the app when data is changed.
```
I suggest that we remove the validation of user input part, as it makes little sense to leave this to UI components in our context